buildkite/resource_test_suite_team.go

```diff
@@ -0,0 +1,203 @@
+	TeamID      types.String `tfsdk:"team_id"`
+	AccessLevel types.String `tfsdk:"access_level"`
+}
+
+type testSuiteTeamResource struct {
+	client *Client
+}
+
+func newTestSuiteTeamResource() resource.Resource {
+	return &testSuiteTeamResource{}
+}
+
+func (*testSuiteTeamResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_test_suite_team"
+}
+
+func (tst *testSuiteTeamResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	tst.client = req.ProviderData.(*Client)
+}
+
+func (tst *testSuiteTeamResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "A test suite team links a collection of tests (suite) to a particular team.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"uuid": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"test_suite_id": schema.StringAttribute{
+				Required: true,
+			},
+			"team_id": schema.StringAttribute{
+				Required: true,
+			},
+			"access_level": schema.StringAttribute{
+				Required: true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("MANAGE_AND_READ", "READ_ONLY"),
+				},
+			},
+		},
+	}
+}
+
+func (tst *testSuiteTeamResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var state testSuiteTeamModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Adding team %s to test suite %s ...", state.TeamID.ValueString(), state.TestSuiteId.ValueString())
+	apiResponse, err := createTestSuiteTeam(
+		tst.client.genqlient,
+		state.TeamID.ValueString(),
+		state.TestSuiteId.ValueString(),
+		SuiteAccessLevels(state.AccessLevel.ValueString()),
+	)
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to create test suite team",
+			fmt.Sprintf("Unable to create test suite team: %s", err.Error()),
+		)
+		return
+	}
+
+	// Set ID and UUID to state
+	state.ID = types.StringValue(apiResponse.TeamSuiteCreate.TeamSuite.Id)
+	state.UUID = types.StringValue(apiResponse.TeamSuiteCreate.TeamSuite.TeamSuiteUuid)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (tst *testSuiteTeamResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state testSuiteTeamModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Reading test suite team with ID %s ...", state.ID.ValueString())
```

Comment:
When does this ever get output?

Reply:
TF_LOG at INFO and higher if its set. More to catch cases if thats set 👍